### ISSUE TITLE
!windows.has(p_window) error on dropdown open

### ISSUE THREAD

**Issue by gitcatrat** [godotengine/godot#46370]
**Godot version:** master

**OS/device including version:** macOS Catalina 10.15.7, Radeon Pro 555, Vulkan

**Issue description:**

Sometimes (not every single time) I get the following error when editor dropdown opens. Unfortunately I haven't figured out the exact state of the world for it to fire besides that single dropdown never fires 2 errors in a row. For example _it seems_ that mesh selection dropdown error fires again if you close the inspector/deselect object and select it again.

List of dropdowns I've found so far:

* Basically all dropdown in inspector, e.g mesh selection, process mode, material override, etc
* Menu dropdowns, e.g help, editor, etc
* Dropdown above the preview, e.g transform, view

```
platform/osx/display_server_osx.mm:2519 - Condition "!windows.has(p_window)" is true. Returning: -1
```

**Steps to reproduce:**
Open the project and start opening listed dropdowns. You should see the error in output.

**Minimal reproduction project:**

[issue.zip](https://github.com/godotengine/godot/files/6035544/issue.zip)

**Comment by Pete-The-Penguin** [comment]
i've had the same problem on kubuntu, so i don't think its just a macos problem.

**Comment by HaSa1002** [comment]
No. It is a bug in the logic calling said code. See referenced PR

**Comment by Pete-The-Penguin** [comment]
ah okay
